meteor/server/api/rest/api.ts

```diff
@@ -0,0 +1,660 @@
+			{ ...this, connection: connection },
+			REST_API_USER_EVENT,
+			getCurrentTime(),
+			rundownPlaylistId,
+			() => {
+				check(rundownPlaylistId, String)
+			},
+			StudioJobs.ResetRundownPlaylist,
+			{
+				playlistId: rundownPlaylistId,
+			}
+		)
+	}
+	async setNextSegment(
+		rundownPlaylistId: RundownPlaylistId,
+		segmentId: SegmentId,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<void>> {
+		return ServerClientAPI.runUserActionInLogForPlaylistOnWorker(
+			{ ...this, connection: connection },
+			REST_API_USER_EVENT,
+			getCurrentTime(),
+			rundownPlaylistId,
+			() => {
+				check(rundownPlaylistId, String)
+				check(segmentId, String)
+			},
+			StudioJobs.SetNextSegment,
+			{
+				playlistId: rundownPlaylistId,
+				nextSegmentId: segmentId,
+			}
+		)
+	}
+	async setNextPart(
+		rundownPlaylistId: RundownPlaylistId,
+		partId: PartId,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<void>> {
+		return ServerClientAPI.runUserActionInLogForPlaylistOnWorker(
+			{ ...this, connection: connection },
+			REST_API_USER_EVENT,
+			getCurrentTime(),
+			rundownPlaylistId,
+			() => {
+				check(rundownPlaylistId, String)
+				check(partId, String)
+			},
+			StudioJobs.SetNextPart,
+			{
+				playlistId: rundownPlaylistId,
+				nextPartId: partId,
+			}
+		)
+	}
+
+	async take(
+		rundownPlaylistId: RundownPlaylistId,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<void>> {
+		triggerWriteAccess()
+		const rundownPlaylist = RundownPlaylists.findOne(rundownPlaylistId)
+		if (!rundownPlaylist) throw new Error(`Rundown playlist ${rundownPlaylistId} does not exist`)
+
+		return ServerClientAPI.runUserActionInLogForPlaylistOnWorker(
+			{ ...this, connection: connection },
+			REST_API_USER_EVENT,
+			getCurrentTime(),
+			rundownPlaylistId,
+			() => {
+				check(rundownPlaylistId, String)
+			},
+			StudioJobs.TakeNextPart,
+			{
+				playlistId: rundownPlaylistId,
+				fromPartInstanceId: rundownPlaylist.currentPartInstanceId,
+			}
+		)
+	}
+
+	async switchRouteSet(studioId: StudioId, routeSetId: string, state: boolean, connection: Meteor.Connection) {
+		return ServerClientAPI.runUserActionInLog(
+			{ ...this, connection: connection },
+			REST_API_USER_EVENT,
+			getCurrentTime(),
+			'switchRouteSet',
+			[studioId, routeSetId, state],
+			async () => {
+				check(studioId, String)
+				check(routeSetId, String)
+				check(state, Boolean)
+
+				const access = await StudioContentWriteAccess.routeSet(this, studioId)
+				return ServerPlayoutAPI.switchRouteSet(access, routeSetId, state)
+			}
+		)
+	}
+}
+registerClassToMeteorMethods(RestAPIMethods, ServerRestAPI, false)
+
+const koaRouter = new KoaRouter()
+
+koaRouter.get('/', async (ctx, next) => {
+	ctx.type = 'application/json'
+	ctx.body = ClientAPI.responseSuccess(await MeteorCall.rest.index())
+	ctx.status = 200
+	await next()
+})
+
+koaRouter.post('/activate/:playlistId', async (ctx, next) => {
```

Comment:
All these URLs don't seem to follow the REST standard.
In REST you design the API around the "resources" (entities etc) in your app, not the operations of the app. 
So in a REST API this URL would be `/playlists/{playlistId}/activate`.

Also, if you have several ids in your URL you still need to specify the resources of those ids.
Further down you have this URL: `/setNextPart/:playlistId/:partId`. Following REST this should be `/playlists/{playlistId}/parts/{partId}/setNext` (Notice that I don't need to write `setNextPart` because that information is now in the URL.
For the Segment equivalent it would be `/playlists/{playlistId}/segments/{segmentId}/setNext`

https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design#organize-the-api-design-around-resources

Reply:
I see your point, but `setNext` is not an operation on a Part but rather an operation that is applied to a specified Part. The `Part` resource doesn't own the operation, it's a resource consumed by the operation.

Perhaps the URL should just be `/setNextPart` with all Ids moved to the body as the operation arguments?

Reply:
I think that doing things like `/playlists/{playlistId}/activate` would be reasonable, and also `/playlists/{playlistId}/setNextPart`, with the partId in the body. 
And I guess the rest way to clear the nextPart would be to do a DELETE instead of a POST with an id of `null`?

Reply:
New definitions in https://github.com/nrkno/sofie-core/pull/777/commits/3286b9f3eadaec9dd3c5afc34957a0c300792544 implemented in https://github.com/nrkno/sofie-core/pull/777/commits/66d52dc36c0b2e4b0bcf768261fd35fe9218aa9b

Reply:
My main point was that you have made a "REST" API, but you don't follow the conventions of REST. So it's basically just an API. If you want to continue to call it a REST API, you should follow the REST conventions